### PR TITLE
Implemented SNS event verification in E2E tests

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-1669

### *What is the problem we're trying to solve*

Verify that the correct Sns event message gets raised for the corresponding API call.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

**Review by humulla — APPROVED** [review]

File: TenureInformationApi.Tests/SnsEventVerifier.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+namespace TenureInformationApi.Tests
+{
+    // TODO - move somewhere common?
+
+    /// <summary>
+    /// Helper class used to verify that the correct Sns event gets raised during an E2E test
+    /// </summary>
+    /// <typeparam name="T">The event class used to create the event.</typeparam>
+    public class SnsEventVerifier<T> : IDisposable where T : class
```
Comment (amidomichael) [inline TenureInformationApi.Tests/SnsEventVerifier.cs:16]: one for `Hackney.Core.Sns.Tests` ?

File: TenureInformationApi.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs [hunk 2]
```diff
@@ -11,20 +8,20 @@ namespace TenureInformationApi.Tests.V1.E2ETests.Stories
 {
     [Story(
        AsA = "Internal Hackney user (such as a Housing Officer or Area housing Manager)",
-       IWant = "the ability to capture a new tenure",
-       SoThat = "I can create a new tenure in the system with all the relevant information")]
+       IWant = "the ability to add a person to a tenure",
+       SoThat = "I can create a link between a tenure and a person")]
     [Collection("Aws collection")]
-    public class PatchTenureTests : IDisposable
+    public class AddPersonToTenureTests : IDisposable
```
Comment (amidomichael) [inline TenureInformationApi.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs:14]: confused by this -I'm guessing the old tests were named incorrectly?

**Review by amidomichael — APPROVED** [review]

File: TenureInformationApi.Tests/SnsEventVerifier.cs (hunk 1 above)
Comment (mikemoody-amido) [reply TenureInformationApi.Tests/SnsEventVerifier.cs:16]: Certainly one for a common package I agree.

File: TenureInformationApi.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs (hunk 2 above)
Comment (mikemoody-amido) [reply TenureInformationApi.Tests/V1/E2ETests/Stories/AddPersonToTenureTests.cs:14]: Yeah... some of the classes for the AddPersonToTenure code were named as if they were for general patch processing.

**Pull request merged**